sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClientTest.java

```diff
@@ -241,73 +368,189 @@ public void recognizeContentInvalidSourceUrl(HttpClient httpClient, FormRecogniz
+        client = getFormRecognizerAsyncClient(httpClient, serviceVersion);
+        customFormDataRunner((data, dataLength) -> beginTrainingLabeledRunner((trainingFilesUrl, useTrainingLabels) -> {
+            SyncPoller<OperationResult, CustomFormModel> syncPoller =
+                getFormTrainingAsyncClient(httpClient, serviceVersion).beginTraining(trainingFilesUrl,
+                    useTrainingLabels).getSyncPoller();
+            syncPoller.waitForCompletion();
 
             assertThrows(RuntimeException.class, () -> client.beginRecognizeCustomForms(
-                new RecognizeCustomFormsOptions(toFluxByteBuffer(data), CUSTOM_FORM_FILE_LENGTH,
-                    null).setFormContentType(FormContentType.APPLICATION_PDF)
-                    .setIncludeTextContent(true)).getSyncPoller());
+                new RecognizeCustomFormsOptions((InputStream) null, dataLength,
+                syncPoller.getFinalResult().getModelId()).setFormContentType(FormContentType.APPLICATION_PDF)
+                .setIncludeTextContent(true)).getSyncPoller());
         }));
     }
 
+    /**
+     * Verifies an exception thrown for a document using null model id.
+     */
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.ai.formrecognizer.TestUtils#getTestParameters")
+    public void recognizeCustomFormLabeledDataWithNullModelId(HttpClient httpClient,
+        FormRecognizerServiceVersion serviceVersion) {
+        client = getFormRecognizerAsyncClient(httpClient, serviceVersion);
+        customFormDataRunner((data, dataLength) -> {
+            Exception ex = assertThrows(RuntimeException.class, () -> client.beginRecognizeCustomForms(
+                new RecognizeCustomFormsOptions(toFluxByteBuffer(data), dataLength,
+                    null).setFormContentType(FormContentType.APPLICATION_PDF).setIncludeTextContent(true))
+                .getSyncPoller());
+            assertEquals(EXPECTED_MODEL_ID_IS_REQUIRED_EXCEPTION_MESSAGE, ex.getMessage());
+        });
+    }
+
+    /**
+     * Verifies an exception thrown for an empty model id when recognizing custom form from URL.
+     */
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.ai.formrecognizer.TestUtils#getTestParameters")
+    public void recognizeCustomFormLabeledDataWithEmptyModelId(HttpClient httpClient,
+        FormRecognizerServiceVersion serviceVersion) {
+        client = getFormRecognizerAsyncClient(httpClient, serviceVersion);
+        customFormDataRunner((data, dataLength) -> {
+            Exception ex = assertThrows(RuntimeException.class, () -> client.beginRecognizeCustomForms(
+                new RecognizeCustomFormsOptions(toFluxByteBuffer(data), dataLength, "")
+                    .setFormContentType(FormContentType.APPLICATION_PDF).setIncludeTextContent(true)).getSyncPoller());
+            assertEquals(EXPECTED_INVALID_UUID_EXCEPTION_MESSAGE, ex.getMessage());
```

Comment:
@maririos @kristapratico Wanted to confirm the exception we are throwing when an empty model Id is passed in respective languages

Reply:
For .NET, if modelId is null we throw `ArgumentNullException` if it is empty, we throw `ArgumentException`.

Reply:
For Python, this is a built-in exception - `ValueError`